lettuce/io.py

```diff
@@ -22,12 +26,36 @@ def write_image(filename, array2d):
     ax.get_yaxis().set_visible(False)
     plt.savefig(filename)
 
+def write_png(filename, array2d):
+    pass
 
-def write_vtk(filename, grid, field_data):
-    # https://vtk.org/Wiki/VTK/Writing_VTK_files_using_python
-    # https://anaconda.org/e3sm/evtk
-    # https://pypi.org/project/pyevtk/
-    raise NotImplementedError
+
+def write_vtk(filename, res, u, p, t):
```

Comment:
1. Can you work this out without making so many copies of the data? (as you know, memory is a precious good in LBM simulations and I don't really see why ux and uy are needed). To blow up the shape you can use `u[:,:,[0]]` and `u[:,:,[1]]` -- or alternatively `u[:,:,1][:,:,None]`
2. I don't think you need the resolution as an argument. Can't you just infer it from the shapes of p and u?
3. Can you make this work for arbitrary dimension (dim = u.shape[0])?